### PR TITLE
Memory leak buster

### PR DESCRIPTION
**Pull request by nayooti**
This PR fixes some memory leaks. 

- `WelcomeViewController` in AppCoordinator is `optional` and is created and set to nil on the fly.
-  `WelcomeViewController`'s `QRReaderController` optional
- fixed retain cycle between WelcomeViewController and ProgressAlertHandler

### CODE REVIEW

File: deltachat-ios/Controller/WelcomeViewController.swift [hunk 1]
```diff
@@ -17,40 +17,19 @@ class WelcomeViewController: UIViewController, ProgressAlertHandler {
 
     private lazy var welcomeView: WelcomeContentView = {
         let view = WelcomeContentView()
-        view.onLogin = {
-            [unowned self] in
-            self.coordinator?.presentLogin()
+        view.onLogin = { [unowned self] in
+            self.coordinator?.showLogin()
```
Comment (cyBerta) [inline deltachat-ios/Controller/WelcomeViewController.swift:21]: the coordinator protocol's method name is presentLogin()

Comment (cyBerta) [reply deltachat-ios/Controller/WelcomeViewController.swift:21]: (the project doesn't compile like that)

File: deltachat-ios/Handler/ProgressAlertHandler.swift [hunk 2]
```diff
@@ -2,34 +2,46 @@ import UIKit
 import DcCore
 
 protocol ProgressAlertHandler: UIViewController {
-    var progressAlert: UIAlertController { get }
-    var configureProgressObserver: Any? { get set }
-    func showProgressAlert(title: String)
+    var progressAlert: UIAlertController? { get set }   // needs to be implemented as weak
```
Comment (r10s) [inline deltachat-ios/Handler/ProgressAlertHandler.swift:5]: please add the the comment _why_ this is needed. at least to me this is not obvious.

File: deltachat-ios/Controller/WelcomeViewController.swift [hunk 3]
```diff
@@ -17,40 +17,19 @@ class WelcomeViewController: UIViewController, ProgressAlertHandler {
...
-        let alert = UIAlertController(title: "", message: "", preferredStyle: .alert)
-        alert.addAction(UIAlertAction(
-            title: String.localized("cancel"),
-            style: .cancel,
-            handler: { _ in
-                self.dcContext.stopOngoingProcess()
-        }))
-        return alert
-    }()
+    private var qrCordeReader: QrCodeReaderController?
```
Comment (r10s) [inline deltachat-ios/Controller/WelcomeViewController.swift:30]: typo in `Corde`

File: deltachat-ios/Controller/WelcomeViewController.swift [hunk 4]
```diff
@@ -17,40 +17,19 @@ class WelcomeViewController: UIViewController, ProgressAlertHandler {
...
         let view = WelcomeContentView()
-        view.onLogin = {
-            [unowned self] in
-            self.coordinator?.presentLogin()
+        view.onLogin = { [unowned self] in
+            self.coordinator?.showLogin()
         }
-        view.onScanQRCode = {
-            [unowned self] in
+        view.onScanQRCode  = { [unowned self] in
```
Comment (r10s) [inline deltachat-ios/Controller/WelcomeViewController.swift:23]: i think, this reformatting (and the other ones, including whitespace changes) has nothing to do with the gist of this pr.

no need to change this now, however, for the future, please keep things in a pr more focused, esp. when a pr is about relatively complicated things as this one.

**Pull request merged**